imap_upstream.adoc

```diff
@@ -31,4 +31,156 @@ The available balancing algorithms are as follows:
+----
+
+== Files that are installed for zimbra-imapd
+
+These are the files that are specifically installed by the
+zimbra-imapd DEB/RPM:
+
+* `/opt/zimbra/bin/zmimapdctl`
+* `/opt/zimbra/conf/imapd.log4j.properties`
+* `/opt/zimbra/lib/jars/oauth-1.4.jar`
+
+In addition, the service makes use of other jar files that are already
+installed into `/opt/zimbra/lib/jars` via the `zimbra-core` package.
+
+== Files that are created during installation
+
+`zmcertmgr` is invoked by `zmsetup.pl` to create the following
+SSL-related files:
+
+* `imapd.crt`
+* `imapd.key`
+* `imapd.keystore`
+
+== Log files
+
+The following log files (in `/opt/zimbra/log`) are created by the
+`zimbra-imapd` service via configuration contained in
+`/opt/zimbra/conf/imapd.log4j.properties`.
+
+* `imapd-audit.log`
+* `imapd.log`
+
+
+== Process Files
+
+The following files are created directly by the `zmimapdctl` script:
+
+* `imapd.out`
+* `imapd.pid`
+
+== Other Files
+
+Activity statistics are collected in `/opt/zimbra/zmstat/imapd.csv`.
+
+
+= Migrating to an imapd pool
+
+When migrating to dedicated pool of IMAP servers, several steps must
+be followed to ensure a smooth migration process. Failure to follow
+these steps may result in service interruptions in the form of dropped
+connections of existing IMAP clients.
+
+== Memcached considerations
+
+Prior to changing any configuration, it should be decided how existing
+routes in `memcached` will be handled. By default, routes are cached in
+`memcached` for 1 day, as specified by the
+`zimbraMemcachedClientExpirySeconds` LDAP attribute. If untouched, it
+will take this long for the cached routes to expire and for the lookup
+extension to send IMAP traffic to the newly provisioned imapd
+servers. There are two things that can be done to change this
+behavior:
+
+1. Flush `memcached` after the imapd pool configuration is
+complete. This will cause all existing IMAP sessions to be restarted
+on the new servers.
+2. One day prior to switching to the imapd pool, modify the value
+`zimbraMemcachedClientExpirySeconds` of to a shorter interval, such as
+30 minutes, and reload the `memcached` configuration with the command
+`zmprov rmcc all`. Doing so will allow embedded IMAP to be turned off
+within a shorter timeframe without risking service interruptions.
+
+== Migration steps
+
+1. Set up the pool of imapd servers. This pool should be sized
+appropriately to accommodate expected IMAP traffic levels. It is
+recommended that you test each server to ensure that it is functioning
+correctly; this can be done via command line with openssl, or by
+configuring an existing IMAP client to point directly at the
+server. These servers should not be listed in the
+`zimbraReverseProxyUpstreamImapServers` attribute yet.
+2. (optional) Specify a load-balancing algorithm using the
+`zimbraImapLoadBalancingAlgorithm` attribute. The default is
+`ClientIPHash`, but custom load-balancing algorithms can be specified as
+well.
```

Comment:
Agreed!  Have updated the wording in a fix-up commit.  I did not regenerate the HTML for the `gh-pages` view yet.